app/controllers/user_blocks_controller.rb

```diff
@@ -6,44 +6,43 @@ def show
     skip_authorization
 
     if current_user.blocking?(params[:blocked_id].to_i)
-      render plain: "blocking"
+      render json: { result: "blocking" }
     else
-      render plain: "not-blocking"
+      render json: { result: "not-blocking" }
     end
   end
 
   def create
-
     authorize UserBlock
     @user_block = UserBlock.new(permitted_attributes(UserBlock))
     @user_block.blocker_id = current_user.id
     @user_block.config = "default"
 
     if @user_block.save
-      block_chat_channel(@user_block.blocked_id)
+      UserBlocks::ChannelHandler.new(@user_block).block_chat_channel
       current_user.stop_following(@user_block.blocked)
       @user_block.blocked.stop_following(current_user)
-      render json: { outcome: "blocked" }
+      render json: { result: "blocked" }
     else
-      render json: { outcome: "error: #{@user_block&.errors&.full_messages}" }
+      render json: { result: "error: #{@user_block.errors.full_messages}" }
```

Comment:
being this an error state, I suggest using error messages like in the API, and adding and error HTTP code.

So something like:

```ruby
render json: { error: @user_block.errors.full_messages.join(", ") }, status: :unprocessable_entity
```

It's also easier in JS to differentiate between a normal result and an error result if they have a different http status

Reply:
Ah yes, forgot to do this. Thanks!